src/compiler/checker.ts

```diff
@@ -7154,35 +7154,36 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                     const methodDeclaration = signatureToSignatureDeclarationHelper(signature, SyntaxKind.MethodSignature, context, { name: propertyName, questionToken: optionalToken }) as MethodSignature;
                     typeElements.push(preserveCommentsOn(methodDeclaration));
                 }
+                if (signatures.length || !optionalToken) {
+                    return;
+                }

```

Comment:
this is the fix, the rest is just dedented because now we fallthrough to the rest of the logic in this case

Reply:
To that end, handy better diff link: https://github.com/microsoft/TypeScript/pull/53938/files?w=1